data/index.js

```diff
@@ -17,18 +17,31 @@ const initialReducer = () => ( {} );
 const store = createStore( initialReducer, {}, flowRight( enhancers ) );
 
 /**
- * Registers a new sub reducer to the global state
+ * Registers a new sub reducer to the global state and returns a Redux-like store object.
  *
- * @param {String} key     Reducer key
- * @param {Object} reducer Reducer function
+ * @param {String}  key     Reducer key
+ * @param {Object}  reducer Reducer function
+ *
+ * @return {Object}         Store Object
  */
 export function registerReducer( key, reducer ) {
 	reducers[ key ] = reducer;
 	store.replaceReducer( combineReducers( reducers ) );
-}
 
-export const subscribe = store.subscribe;
+	return {
+		dispatch: store.dispatch,
+		getState: () => get( store.getState(), key ),
+		subscribe( listener ) {
```

Comment:
I'm trying to understand why using `store.subscribe` is not enough here. When I look at Redux source code, I see that `listener` is executed only at the end of the `dispatch` call: https://github.com/reactjs/redux/blob/master/src/createStore.js#L194. This code tries to filter out all listener calls when the local state hasn't changed. It seems like a good idea, but the question is how it works in practice? Is Redux really immutable here? It might be not architected this way to speed up building new state.

Reply:
The more we'd have module and plugins using the "data" moodule, the more we'd have actions handled only by a separate reducer which would create unecessary listener calls for most modules/plugins. I think this is a necessary performance improvement. 

Not noticeable right now because we only have one registered reducer.

Reply:
Okey, I looked into `combineReducers` code. It seems to be smart enough to return old state when nothing has changed. See: https://github.com/reactjs/redux/blob/master/src/combineReducers.js#L118.

Reply:
Yes, but in this case the state did change but only for one module.